pkg/client/restclient/request.go

```diff
@@ -712,26 +721,33 @@ func (r *Request) Stream() (io.ReadCloser, error) {
 
 	r.tryThrottle()
 
-	url := r.URL().String()
-	req, err := http.NewRequest(r.verb, url, nil)
-	if err != nil {
-		return nil, err
-	}
-	req.Header = r.headers
-	client := r.client
-	if client == nil {
-		client = http.DefaultClient
-	}
-	r.backoffMgr.Sleep(r.backoffMgr.CalculateBackoff(r.URL()))
-	resp, err := client.Do(req)
-	updateURLMetrics(r, resp, err)
-	if r.baseURL != nil {
-		if err != nil {
-			r.backoffMgr.UpdateBackoff(r.URL(), err, 0)
-		} else {
-			r.backoffMgr.UpdateBackoff(r.URL(), err, resp.StatusCode)
-		}
-	}
+	var req *http.Request
+	var resp *http.Response
+	err := r.urlProvider.WrapWithRetry(
+		func() error {
```

Comment:
This block now looks identical across all the places it's called here.  Is that true?

If so please leave a TODO in the code that this needs to be refactored to be the same, and that req/resp should be returned values from the function, not closure parameters.